sonar/resources/projects/permissions.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Swiss Open Access Repository
+# Copyright (C) 2021 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Projects permission policy."""
+
+from elasticsearch_dsl.query import Q
+from flask import request
+from invenio_access.permissions import any_user
+from invenio_records_permissions import RecordPermissionPolicy
+from invenio_records_permissions.generators import Admin
+
+from sonar.modules.api import SonarRecord
+from sonar.modules.documents.api import DocumentRecord
+from sonar.modules.organisations.api import current_organisation
+from sonar.modules.users.api import current_user_record
+
+
+class Read(Admin):
+    """Read projects permissions."""
+
+    def query_filter(self, **kwargs):
+        """Search filters."""
+        if current_user_record.is_superuser:
+            return Q('match_all')
+
+        # For admin and moderator, only records that belongs to his
+        # organisation. The same rule is applied when searching project in
+        # typeahead input.
+        # TODO: Find a better way for handling typeahead calls..
+        if current_user_record.is_moderator or (
+                request.args.get('q') and
+                request.args['q'].startswith('autocomplete_name')):
+            return Q('term',
+                     metadata__organisation__pid=current_organisation['pid'])
+
+        # For user, only records that belongs to him.
+        if current_user_record.is_submitter:
```

Comment:
You can use here a current_user.roles to avoid to retrieve the current_user_record.

Reply:
I think `current_user_record` is needed because I need the record `pid` (https://github.com/rero/sonar/pull/508/files#diff-fbe4e99e5d9b0b42c6473bb695fb025bd08928ed9f67a6a39c7d1e1d3dfef31aR52)